iterative/utils/analytics.go

```diff
@@ -223,26 +233,59 @@ func JitsuEventPayload(action string, e error, d *schema.ResourceData) map[strin
 	return payload
 }
 
-func SendJitsuEvent(action string, e error, d *schema.ResourceData) {
-	if d == nil {
-		return
+func SendJitsuEvent(action string, e error, extra map[string]interface{}) {
+	for _, env := range []string{"DO_NOT_TRACK", "DVC_NO_ANALYTICS"} {
+		if _, ok := os.LookupEnv(env); ok {
+			logrus.Debugf("analytics: %s environment variable is set; doing nothing", env)
+			return
+		}
 	}
 
-	if _, ok := os.LookupEnv("DO_NOT_TRACK"); ok {
-		logrus.Debug("Analytics: DO_NOT_TRACK enabled")
-		return
-	}
+	wg.Add(1)
+	ctx, _ := context.WithTimeout(context.Background(), Timeout)
```

Comment:
Without calling the returned cancel func doesn't this technically leak the context resources if the goroutine finishes before the timeout?

I don't think it much of a real issue since it will generally be terminating right after, ie this is never a long running process.

Reply:
Resolved with https://github.com/iterative/terraform-provider-iterative/pull/538/commits/b305ad58e43d8731d1eb04ad41a835ac547f2d97 and getting rid of contexts. You're right about “leaking” the context resources for exactly 5 seconds, even if the goroutine finished a bit before.

Reply:
Definitely not an issue, but we weren't using the cancel function anyway, so it's better to get rid of it in favor of the native timeout.